xmodule/x_module.py

```diff
@@ -1243,7 +1244,7 @@ def resource_url(self, resource):
         raise NotImplementedError("edX Platform doesn't currently implement XBlock resource urls")
 
     def add_block_as_child_node(self, block, node):
-        child = etree.SubElement(node, "unknown")
+        child = etree.SubElement(node, block.category)
```

Comment:
Moved from an old `export_to_xml`. Without this change, an XBlock that is not installed is exported as an "unknown" XML tag.

Reply:
One edge case that we've had issues with in the past is what happens when you have this scenario:

1. Custom XBlock exists. For this example, let's call it a "MusicBlock", with a tag of "music".
2. A course with `<music>` is imported.
3. MusicBlock is uninstalled.
4. A course with `<music>` is exported.

If MusicBlock had never been installed, it goes to the catch-all and the XML is spit out the way it came in. But in this case, it was installed so the original XML was never preserved–it was deserialized into whatever data MusicBlock decided it should be parsed into. In the past, we used to try to look up MusicBlock and fail the export because that code couldn't be found. The fix at one point was to just not export the content at all.

I'm not sure what the behavior of this is with the current codebase. If the existing code on master breaks in this scenario today, you don't have to fix it here. But I would like to make sure that this commit doesn't introduce a regression and that if the above scenario works on master today, this commit won't break it.